diff --git a/__init__.py b/__init__.py
index c43cc2eb7e33..aa8544830a6a 100644
--- a/__init__.py
+++ b/__init__.py
@@ -11,7 +11,6 @@ import lz4.block
 import datetime
 import configparser
 import base64
-from Crypto.Cipher import AES
 
 try:
     import json
@@ -24,10 +23,6 @@ try:
 except ImportError:
     import sqlite3
 
-# external dependencies
-import keyring
-import pyaes
-from pbkdf2 import PBKDF2
 
 __doc__ = 'Load browser cookies into a cookiejar'
 
@@ -119,6 +114,7 @@ def get_linux_pass(browser="Chrome"):
     except (ImportError, AttributeError):
         pass
     else:
+        import keyring
         flags = Secret.ServiceFlags.LOAD_COLLECTIONS
         service = Secret.Service.get_sync(flags)
 
@@ -161,7 +157,11 @@ class Chrome:
         self.length = 16
         # domain name to filter cookies by
         self.domain_name = domain_name
+
+        from pbkdf2 import PBKDF2
+
         if sys.platform == 'darwin':
+            import keyring
             # running Chrome on OSX
             my_pass = keyring.get_password('Chrome Safe Storage', 'Chrome').encode(
                 'utf8')  # get key from keyring
@@ -306,6 +306,8 @@ class Chrome:
     def _decrypt(self, value, encrypted_value):
         """Decrypt encoded cookies
         """
+        from Crypto.Cipher import AES
+        import pyaes
 
         if sys.platform == 'win32':
             try:
